internal/monitor/traceexporter.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"context"
+
+	cloudtrace "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/trace"
+	"github.com/googlecloudplatform/gcsfuse/v2/cfg"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+
+	"go.opentelemetry.io/contrib/detectors/gcp"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+type ShutdownFn func(ctx context.Context) error
+
+// SetupTracing bootstraps the OpenTelemetry tracing pipeline.
+func SetupTracing(ctx context.Context, c *cfg.Config) ShutdownFn {
+	tp, shutdown, err := newTraceProvider(ctx, c)
+	if err == nil && tp != nil {
+		otel.SetTracerProvider(tp)
+		return shutdown
+	}
+	return nil
+}
+
+func newTraceProvider(ctx context.Context, c *cfg.Config) (trace.TracerProvider, ShutdownFn, error) {
+	switch c.Monitoring.ExperimentalTracingMode {
```

Comment:
Should there be a case here where ExperimentalTracingMode is not defined
Or c.Monitoring itself null.

Reply:
c.Monitoring cannot be nil. It'll be empty but never nil. Same goes for ExperimentalTracingMode as well.